pkg/skaffold/debug/transform.go

```diff
@@ -122,16 +125,42 @@ func transformPodSpec(metadata *metav1.ObjectMeta, podSpec *v1.PodSpec, retrieve
 	}
 	// containers are required to have unique name within a pod
 	configurations := make(map[string]map[string]interface{})
+	requiredSupportImages := make(map[string]string)
+	var supportFilesRequired []*v1.Container
 	for i := range podSpec.Containers {
 		container := &podSpec.Containers[i]
 		// we only reconfigure build artifacts
-		if configuration, err := transformContainer(container, retrieveImageConfiguration, portAlloc); err == nil {
+		if configuration, requiredSupportImage, err := transformContainer(container, retrieveImageConfiguration, portAlloc); err == nil {
 			configurations[container.Name] = configuration
+			if len(requiredSupportImage) > 0 {
+				logrus.Infof("%q requires debugging support image %q", container.Name, requiredSupportImage)
+				supportFilesRequired = append(supportFilesRequired, container)
+				requiredSupportImages[requiredSupportImage] = requiredSupportImage
+			}
 			// todo: add this artifact to the watch list?
 		} else {
-			logrus.Infof("Image [%s] not configured for debugging: %v", container.Image, err)
+			logrus.Infof("Image %q not configured for debugging: %v", container.Name, err)
 		}
 	}
+	if len(supportFilesRequired) > 0 {
+		logrus.Infof("Configuring installation of debugging support files")
+		supportVolume := v1.Volume{Name: "debugging-support-files", VolumeSource: v1.VolumeSource{EmptyDir: &v1.EmptyDirVolumeSource{}}}
+		podSpec.Volumes = append(podSpec.Volumes, supportVolume)
+		supportVolumeMount := v1.VolumeMount{Name: "debugging-support-files", MountPath: "/dbg"}
+		for _, container := range supportFilesRequired {
+			container.VolumeMounts = append(container.VolumeMounts, supportVolumeMount)
```

Comment:
Should this be done in the` transform.Apply`?

Reply:
There will be multiple transformers that require debugging support files: Go requires Delve, PHP requires XDEBUG, Python requires `ptvsd`.  I'll add more comments to describe the process.

Reply:
Right. My concern is that this code is transforming a container, not in `transformContainer`, but in `transformPodSpec`.

Reply:
The actual populating of the volume is done here, so it makes sense to add the volume here rather than scatter is amongst the different transformers.  The `transformContainer` just needs to know that the files are available on `/dbg`.